### ISSUE TITLE
Uploading artifacts Failure

### ISSUE THREAD

**Issue by Shabinder** [nexus-actions/create-nexus-staging-repo#1]
![image](https://user-images.githubusercontent.com/44765050/118321766-b035ea80-b51b-11eb-9365-ef449266e407.png)

Creation of repo seems to be successful, but Server Responds with **Bad Request 400** when uploading an artifact, see below image:

![image](https://user-images.githubusercontent.com/44765050/118321974-099e1980-b51c-11eb-834d-683353fbabae.png)
![image](https://user-images.githubusercontent.com/44765050/118322023-1ae72600-b51c-11eb-90ec-e51e055a7b6e.png)

**Comment by Shabinder** [comment]
Repo with workflow and publish task is: https://github.com/Shabinder/youtube-api-dl

**Comment by Shabinder** [comment]
It seems createStaging repo is unable to create a repo and passes blank ID to other jobs.

It would be better if there was error handling and better logging

**Comment by romainbsl** [comment]
Did you check on Sonatype that your repo is not created ?

In fact, there is some error handling:
https://github.com/nexus-actions/create-nexus-staging-repo/blob/0c1b2e32f7b4a5246215d5ac35afd20da24f8d8a/entrypoint.sh#L14

However the documentation is wrong, 
TLDR: there is a typo in `repository-id` it should be `repository_id`.

you should retrieve the job's output with 
```
repository_id: ${{ steps.create.outputs.repository_id }}
```

like in this script
```
jobs:
  create-staging-repository:
    name: Create staging repository
    runs-on: ubuntu-latest
    outputs:
      repository_id: ${{ steps.create.outputs.repository_id }}
    steps:
      - id: create
        uses: nexus-actions/create-nexus-staging-repo@v1.1
        with:
          username: ${{ secrets.SONATYPE_USERNAME }}
          password: ${{ secrets.SONATYPE_PASSWORD }}
          staging_profile_id: ${{ secrets.SONATYPE_PROFILE_ID }}
          description: ${{ github.repository }}/${{ github.workflow }}#${{ github.run_number }}
```

**Comment by Shabinder** [comment]
Yes, I checked and a repo was created successfully in nexus manager, lemme check again after your doing said changes

**Comment by Shabinder** [comment]
Still Same error:
<img width="998" alt="image" src="https://user-images.githubusercontent.com/44765050/118356045-01cc8c80-b591-11eb-9a47-1eb8e25dfa5f.png">

Repository Got created but still same error:
<img width="913" alt="image" src="https://user-images.githubusercontent.com/44765050/118356068-1f015b00-b591-11eb-8aa4-af3664fb6d50.png">

workflow after fixes: 
<img width="921" alt="image" src="https://user-images.githubusercontent.com/44765050/118356101-46f0be80-b591-11eb-8a1c-0b1af84a14ac.png">

**Comment by Shabinder** [comment]
For now I created a workaround , create an empty repo using create_staging_repo beforehand and then pass that as a REPO Secret to both the tasks manually.

**Comment by romainbsl** [comment]
You should consider using an underscore _ instead of the dash - in the job variable as well, we had some issue with this on GitHub Actions...

**Comment by Shabinder** [comment]
lemme try

**Comment by romainbsl** [comment]
You can find a working example here

https://github.com/Kodein-Framework/Kodein-DI/blob/master/.github/workflows/release.yml

**Comment by Shabinder** [comment]
I edited, but still repoID was blank, I couldn't find any issue with my workflow ,

check it out https://github.com/Shabinder/youtube-api-dl/blob/master/.github/workflows/main.yml

**Comment by romainbsl** [comment]
Why are you overriding the base_url ?
When I try it with curl I get a 403 http error.

can you try with `https://oss.sonatype.org/service/local/` ? in your gradle configuration as well ?
I'll try with your URL in a test to see if there is an issue with my script.

**Comment by romainbsl** [comment]
Ok I found the issue !

looking at the output of the job https://github.com/Shabinder/youtube-api-dl/runs/2590468930?check_suite_focus=true

Looking at the **complete job section**, we can see:
`Warning: Skip output 'repository_id' since it may contain secret.`

This means that your org on sonatype is the same as your github secret `secrets.NEXUS_ACTIONS_SONATYPE_USERNAME`, thus Github avoid leaking secrets by omitting it in the ouputs....

**Comment by Shabinder** [comment]
Ohhh, thanks for helping me out here , will create an issue in sonatype's jira to request a change username

**Comment by martinbonnin** [comment]
I got bit by this as well. TBH, I think Github could relax that. Or at least give a huge warning as every time this happens, it means some workflows won't work as expected. 

My solution was to hardcode my sonatype username:

```yaml
    steps:
    - id: create
      uses: nexus-actions/create-nexus-staging-repo@v1
      with:
        # The username you use to connect to Sonatype's Jira
        # Do not use secrets for the username since the repository id will contain it
        # and it will not be passed across jobs :/
        username: mbonnin
```

I think username is ok to share as most of the times it's not really secret.
